src/language/templates/pybind/pyast.hpp

```diff
@@ -63,32 +63,36 @@ struct PyAst: public Ast {
         PYBIND11_OVERLOAD(Ast*, Ast, clone, );
     }
 
-    AstNodeType get_node_type() override {
+    AstNodeType get_node_type() const override {
         PYBIND11_OVERLOAD_PURE(AstNodeType,    // Return type
                                Ast,            // Parent class
                                get_node_type,  // Name of function in C++ (must match Python name)
                                                // No argument (trailing ,)
         );
     }
 
-    std::string get_node_type_name() override {
+    std::string get_node_type_name() const override {
         PYBIND11_OVERLOAD_PURE(std::string, Ast, get_node_type_name, );
     }
 
-    std::string get_node_name() override {
+    std::string get_node_name() const override {
         PYBIND11_OVERLOAD(std::string, Ast, get_node_name, );
     }
 
-    std::string get_nmodl_name() override {
+    std::string get_nmodl_name() const override {
         PYBIND11_OVERLOAD(std::string, Ast, get_nmodl_name, );
     }
 
     std::shared_ptr<Ast> get_shared_ptr() override {
         PYBIND11_OVERLOAD(std::shared_ptr<Ast>, Ast, get_shared_ptr, );
     }
 
-    ModToken* get_token() override {
-        PYBIND11_OVERLOAD(ModToken*, Ast, get_token, );
+    std::shared_ptr<const Ast> get_shared_ptr() const override {
```

Comment:
I had not added this overload because it has no python interface. It didn't really make sense to expose this in python. Any reason you're adding this, or did I miss something?

Reply:
I have just reverted this addition